control/control_origination.go

```diff
@@ -59,7 +59,27 @@ func detectControlOriginKey(textNodes []xml.Node) string {
 	return noOrigin
 }
 
-func newControlOrigination(st SummaryTable) (*controlOrigination, error) {
+func detectControlOriginKeyFromData(text string) string {
+	switch text {
+	case "service_provider_corporate":
+		return serviceProviderCorporateOrigination
+	case "service_provided_system_specific":
+		return serviceProviderSystemSpecificOrigination
+	case "hybrid":
+		return serviceProviderHybridOrigination
+	case "customer_configured":
+		return configuredByCustomerOrigination
+	case "customer_provided":
+		return providedByCustomerOrigination
+	case "shared":
+		return sharedOrigination
+	case "inherited":
+		return inheritedOrigination
+	}
+	return noOrigin
```

Comment:
Can we simplify this and the variables above to simply be a map from the origin key to the document substring?